#! /bin/sh -e
## debian-zaptel.dpatch by Mark Purcell <msp@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Patch asterisk to use installed zaptel

if [ $# -lt 1 ]; then
    echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch} ${2:+-d $2}"

case "$1" in
    -patch) patch -p1 ${patch_opts} < $0;;
    -unpatch) patch -R -p1 ${patch_opts} < $0;;
    *)
        echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
        exit 1;;
esac

exit 0

@DPATCH@
diff -urNad asterisk-1.0.5/Makefile /tmp/dpep.dtA8vi/asterisk-1.0.5/Makefile
--- asterisk-1.0.5/Makefile	2005-02-08 23:54:59.000000000 +0100
+++ /tmp/dpep.dtA8vi/asterisk-1.0.5/Makefile	2005-02-08 23:54:59.000000000 +0100
@@ -124,8 +124,7 @@
 #Uncomment this to use the older DSP routines
 #CFLAGS+=-DOLD_DSP_ROUTINES
 
-CFLAGS+=$(shell if [ -f /usr/include/linux/zaptel.h ]; then echo "-DZAPTEL_OPTIMIZATIONS"; fi)
-CFLAGS+=$(shell if [ -f /usr/local/include/zaptel.h ]; then echo "-DZAPTEL_OPTIMIZATIONS"; fi)
+CFLAGS+=$(shell if [ -f /usr/src/modules/zaptel/zaptel.h ]; then echo "-DZAPTEL_OPTIMIZATIONS"; fi)
 
 LIBEDIT=-Leditline -ledit
 
diff -urNad asterisk-1.0.5/apps/Makefile /tmp/dpep.dtA8vi/asterisk-1.0.5/apps/Makefile
--- asterisk-1.0.5/apps/Makefile	2005-02-08 23:54:59.000000000 +0100
+++ /tmp/dpep.dtA8vi/asterisk-1.0.5/apps/Makefile	2005-02-08 23:54:59.000000000 +0100
@@ -35,12 +35,11 @@
 APPS+=app_intercom.so
 endif
 
-#APPS+=app_sql_postgres.so
+APPS+=app_sql_postgres.so
 #APPS+=app_sql_odbc.so
 #APPS+=app_rpt.so
 
-APPS+=$(shell if [ -f /usr/include/linux/zaptel.h ]; then echo "app_zapras.so app_meetme.so app_flash.so app_zapbarge.so app_zapscan.so" ; fi)
-APPS+=$(shell if [ -f /usr/local/include/zaptel.h ]; then echo "app_zapras.so app_meetme.so app_flash.so app_zapbarge.so app_zapscan.so" ; fi)
+APPS+=$(shell if [ -f /usr/src/modules/zaptel/zaptel.h ]; then echo "app_zapras.so app_meetme.so app_flash.so app_zapbarge.so app_zapscan.so" ; fi)
 APPS+=$(shell if [ -f /usr/include/osp/osp.h ]; then echo "app_osplookup.so" ; fi)
 
 CFLAGS+=-fPIC -DPIC
diff -urNad asterisk-1.0.5/apps/app_flash.c /tmp/dpep.dtA8vi/asterisk-1.0.5/apps/app_flash.c
--- asterisk-1.0.5/apps/app_flash.c	2004-06-22 21:32:52.000000000 +0200
+++ /tmp/dpep.dtA8vi/asterisk-1.0.5/apps/app_flash.c	2005-02-08 23:54:59.000000000 +0100
@@ -22,7 +22,7 @@
 #include <asterisk/options.h>
 #include <sys/ioctl.h>
 #ifdef __linux__
-#include <linux/zaptel.h>
+#include "/usr/src/modules/zaptel/zaptel.h"
 #else
 #include <zaptel.h>
 #endif /* __linux__ */
diff -urNad asterisk-1.0.5/apps/app_meetme.c /tmp/dpep.dtA8vi/asterisk-1.0.5/apps/app_meetme.c
--- asterisk-1.0.5/apps/app_meetme.c	2004-11-25 19:27:22.000000000 +0100
+++ /tmp/dpep.dtA8vi/asterisk-1.0.5/apps/app_meetme.c	2005-02-08 23:54:59.000000000 +0100
@@ -33,7 +33,7 @@
 #include <sys/ioctl.h>
 
 #ifdef __linux__
-#include <linux/zaptel.h>
+#include "/usr/src/modules/zaptel/zaptel.h"
 #else
 #include <zaptel.h>
 #endif /* __linux__ */
diff -urNad asterisk-1.0.5/apps/app_rpt.c /tmp/dpep.dtA8vi/asterisk-1.0.5/apps/app_rpt.c
--- asterisk-1.0.5/apps/app_rpt.c	2004-09-13 14:12:17.000000000 +0200
+++ /tmp/dpep.dtA8vi/asterisk-1.0.5/apps/app_rpt.c	2005-02-08 23:54:59.000000000 +0100
@@ -135,7 +135,7 @@
 #include <sys/io.h>
 #include <math.h>
 #include <tonezone.h>
-#include <linux/zaptel.h>
+#include "/usr/src/modules/zaptel/zaptel.h"
 
 static  char *tdesc = "Radio Repeater / Remote Base  version 0.17  09/13/2004";
 static char *app = "Rpt";
diff -urNad asterisk-1.0.5/apps/app_zapbarge.c /tmp/dpep.dtA8vi/asterisk-1.0.5/apps/app_zapbarge.c
--- asterisk-1.0.5/apps/app_zapbarge.c	2004-07-14 09:34:34.000000000 +0200
+++ /tmp/dpep.dtA8vi/asterisk-1.0.5/apps/app_zapbarge.c	2005-02-08 23:54:59.000000000 +0100
@@ -34,7 +34,7 @@
 #include <sys/ioctl.h>
 
 #ifdef __linux__
-#include <linux/zaptel.h>
+#include "/usr/src/modules/zaptel/zaptel.h"
 #else
 #include <zaptel.h>
 #endif /* __linux__ */
diff -urNad asterisk-1.0.5/apps/app_zapras.c /tmp/dpep.dtA8vi/asterisk-1.0.5/apps/app_zapras.c
--- asterisk-1.0.5/apps/app_zapras.c	2004-06-22 21:32:52.000000000 +0200
+++ /tmp/dpep.dtA8vi/asterisk-1.0.5/apps/app_zapras.c	2005-02-08 23:54:59.000000000 +0100
@@ -36,7 +36,7 @@
 
 /* Need some zaptel help here */
 #ifdef __linux__
-#include <linux/zaptel.h>
+#include "/usr/src/modules/zaptel/zaptel.h"
 #else
 #include <zaptel.h>
 #endif /* __linux__ */
diff -urNad asterisk-1.0.5/apps/app_zapscan.c /tmp/dpep.dtA8vi/asterisk-1.0.5/apps/app_zapscan.c
--- asterisk-1.0.5/apps/app_zapscan.c	2004-07-14 09:34:34.000000000 +0200
+++ /tmp/dpep.dtA8vi/asterisk-1.0.5/apps/app_zapscan.c	2005-02-08 23:54:59.000000000 +0100
@@ -35,7 +35,7 @@
 #include <sys/ioctl.h>
 
 #ifdef __linux__
-#include <linux/zaptel.h>
+#include "/usr/src/modules/zaptel/zaptel.h"
 #else
 #include <zaptel.h>
 #endif /* __linux__ */
diff -urNad asterisk-1.0.5/channel.c /tmp/dpep.dtA8vi/asterisk-1.0.5/channel.c
--- asterisk-1.0.5/channel.c	2005-01-22 02:44:12.000000000 +0100
+++ /tmp/dpep.dtA8vi/asterisk-1.0.5/channel.c	2005-02-08 23:54:59.000000000 +0100
@@ -41,7 +41,7 @@
 #ifdef ZAPTEL_OPTIMIZATIONS
 #include <sys/ioctl.h>
 #ifdef __linux__
-#include <linux/zaptel.h>
+#include "/usr/src/modules/zaptel/zaptel.h"
 #else
 #include <zaptel.h>
 #endif /* __linux__ */
diff -urNad asterisk-1.0.5/channels/Makefile /tmp/dpep.dtA8vi/asterisk-1.0.5/channels/Makefile
--- asterisk-1.0.5/channels/Makefile	2005-02-08 23:54:59.000000000 +0100
+++ /tmp/dpep.dtA8vi/asterisk-1.0.5/channels/Makefile	2005-02-08 23:54:59.000000000 +0100
@@ -80,8 +80,7 @@
 CFLAGS+=$(shell [ -f alsa-monitor.h ] && echo " -DALSA_MONITOR")
 ZAPPRI=$(shell [ -f /usr/lib/libpri.so.1 ] && echo "-lpri")
 ZAPR2=$(shell [ -f /usr/lib/libmfcr2.so.1 ] && echo "-lmfcr2")
-CFLAGS+=$(shell [ -f /usr/include/linux/zaptel.h ] && echo "-DIAX_TRUNKING")
-CFLAGS+=$(shell [ -f /usr/local/include/zaptel.h ] && echo "-DIAX_TRUNKING")
+CFLAGS+=$(shell [ -f /usr/src/modules/zaptel/zaptel.h ] && echo "-DIAX_TRUNKING")
 CHANNEL_LIBS+=$(shell [ -f /usr/include/vpbapi.h ] && echo "chan_vpb.so" )
 CFLAGS+=$(shell [ -f /usr/include/vpbapi.h ] && echo " -DLINUX")
 
@@ -103,8 +102,7 @@
 
 ZAPDIR=/usr/lib
 
-CHANNEL_LIBS+=$(shell [ -f /usr/include/linux/zaptel.h ] && echo "chan_zap.so")
-CHANNEL_LIBS+=$(shell [ -f /usr/local/include/zaptel.h ] && echo "chan_zap.so")
+CHANNEL_LIBS+=$(shell [ -f /usr/src/modules/zaptel/zaptel.h ] && echo "chan_zap.so")
 
 CHANNEL_LIBS+=$(shell [ -f /usr/include/nbs.h ] && echo "chan_nbs.so" )
 
diff -urNad asterisk-1.0.5/channels/chan_iax2.c /tmp/dpep.dtA8vi/asterisk-1.0.5/channels/chan_iax2.c
--- asterisk-1.0.5/channels/chan_iax2.c	2005-01-04 20:02:16.000000000 +0100
+++ /tmp/dpep.dtA8vi/asterisk-1.0.5/channels/chan_iax2.c	2005-02-08 23:54:59.000000000 +0100
@@ -57,7 +57,7 @@
 #ifdef IAX_TRUNKING
 #include <sys/ioctl.h>
 #ifdef __linux__
-#include <linux/zaptel.h>
+#include "/usr/src/modules/zaptel/zaptel.h"
 #else
 #include <zaptel.h>
 #endif /* __linux__ */
diff -urNad asterisk-1.0.5/channels/chan_zap.c /tmp/dpep.dtA8vi/asterisk-1.0.5/channels/chan_zap.c
--- asterisk-1.0.5/channels/chan_zap.c	2005-02-08 23:54:59.000000000 +0100
+++ /tmp/dpep.dtA8vi/asterisk-1.0.5/channels/chan_zap.c	2005-02-08 23:54:59.000000000 +0100
@@ -46,7 +46,7 @@
 #include <unistd.h>
 #include <sys/ioctl.h>
 #ifdef __linux__
-#include <linux/zaptel.h>
+#include "/usr/src/modules/zaptel/zaptel.h"
 #else
 #include <zaptel.h>
 #endif /* __linux__ */
diff -urNad asterisk-1.0.5/res/Makefile /tmp/dpep.dtA8vi/asterisk-1.0.5/res/Makefile
--- asterisk-1.0.5/res/Makefile	2005-02-08 23:54:59.000000000 +0100
+++ /tmp/dpep.dtA8vi/asterisk-1.0.5/res/Makefile	2005-02-08 23:56:08.000000000 +0100
@@ -20,8 +20,7 @@
 CRYPTO_LIBS=-lssl -lcrypto
 
 CFLAGS+=-fPIC -DPIC
-CFLAGS+=$(shell [ -f /usr/include/linux/zaptel.h ] && echo " -DZAPATA_MOH")
-CFLAGS+=$(shell [ -f /usr/local/include/zaptel.h ] && echo " -DZAPATA_MOH")
+CFLAGS+=$(shell [ -f /usr/src/modules/zaptel/zaptel.h ] && echo " -DZAPATA_MOH")
 #
 # Work around buggy RedHat 9.0
 #
diff -urNad asterisk-1.0.5/res/res_musiconhold.c /tmp/dpep.dtA8vi/asterisk-1.0.5/res/res_musiconhold.c
--- asterisk-1.0.5/res/res_musiconhold.c	2005-01-08 22:54:24.000000000 +0100
+++ /tmp/dpep.dtA8vi/asterisk-1.0.5/res/res_musiconhold.c	2005-02-08 23:54:59.000000000 +0100
@@ -38,7 +38,7 @@
 #include <dirent.h>
 #ifdef ZAPATA_MOH
 #ifdef __linux__
-#include <linux/zaptel.h>
+#include "/usr/src/modules/zaptel/zaptel.h"
 #else
 #include <zaptel.h>
 #endif /* __linux__ */
